CityOfDetroit/bloom
# 1058/updated public listing filters

- Closes [#issue](https://github.com/CityOfDetroit/bloom/issues/1058)

## Description
This modifies the existing filters to match up with the mock in the linked issue

## How the filters work
**Availability filter:**
Vacant Units -> a unit group has availability, if paired with an open/closed waitlist filter option then the unit group must have availability and be marked as waitlist open/closed

open/closed waitlist -> a unit group must be marked as waitlist open/closed, if both are selected this filter does nothing

if a listing has a unit groups where at least one is marked as open waitlist and one is marked as closed waitlist that listing would always show regardless of open/closed/both waitlist options being selected as filters

**Bedroom Size**
this is functionally an or, so if you selected 2 different bedroom size options, a unit group must have at least one of those for the listing to show

**Monthly Rent Range**
this is an and functionality (if both are filled the rent must be between those values) 

**Community Programs**
this is functionally an or, so if you selected 2 different program options, at least one of those must be true on the listing for the listing to show

**Region**
This is functionally an or, if two or more are selected the listing must have at least one of those regions for the listing to show

**Accessibility Features**
This is functionally an and, if two are more are selected the listing must have all of those features for the listing to show

## How Can This Be Tested/Reviewed?
navigate to public site
navigate to the listings page
use the filter button, and verify the filter drawer matches with the mock
verify the filters work

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
